examples/kaniko.groovy

```diff
@@ -0,0 +1,43 @@
+/**
+ * This pipeline will build and deploy a Docker image with Kaniko
+ * https://github.com/GoogleContainerTools/kaniko
+ * without needing a Docker host
+ *
+ * You need to create a jenkins-docker-cfg secret with your docker config
+ * as described in
+ * https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/#create-a-secret-in-the-cluster-that-holds-your-authorization-token
```

Comment:
Those instructions are specifically for pulling images used by a `pod` - not for pushing to a registry. I had to use the following `secret` configuration to make sure the Docker credentials were placed in `~/.docker/config.json`:

```
...
     volumeMounts:
       - name: jenkins-docker-cfg
         mountPath: /root/.docker
   volumes:
     - name: jenkins-docker-cfg
       secret:
         secretName: jenkins-docker-cfg
         items:
         - key: .dockerconfigjson
           path: config.json
```

Reply:
I haven't tried, but I assumed it was the same format, will take a look.

This I have tried https://github.com/GoogleContainerTools/kaniko/pull/129

Reply:
Figured it out, the docker-registry secret type crates a .dockercfg file that needs to be put in root
It does not create the new .docker/config.json format